coordinator/src/services/translation/prescription/index.ts

```diff
@@ -163,33 +166,62 @@ function convertDispensingSitePreference(
   return new prescriptions.DispensingSitePreference(dispensingSitePreferenceValue)
 }
 
-function extractText(fhirCommunicationRequests: Array<fhir.CommunicationRequest>): Array<core.Text> {
+function isContentStringPayload(
+  payload: ContentStringPayload | ContentReferencePayload
+): payload is ContentStringPayload {
+  return !!(payload as ContentStringPayload).contentString
+}
+
+function isContentReferencePayload(
+  payload: ContentStringPayload | ContentReferencePayload
+): payload is ContentReferencePayload {
+  return !!(payload as ContentReferencePayload).contentReference
+}
+
+function extractPatientInfoText(fhirCommunicationRequests: Array<fhir.CommunicationRequest>): Array<core.Text> {
   return fhirCommunicationRequests
     .flatMap(communicationRequest => communicationRequest.payload)
     .filter(isTruthy)
-    .map(contentStringPayload => contentStringPayload.contentString)
-    .filter(isTruthy)
+    .filter(isContentStringPayload)
+    .map(payload => payload.contentString)
     .map(contentString => new core.Text(contentString))
 }
 
+function extractMedicationListText(
+  fhirBundle: fhir.Bundle,
+  fhirCommunicationRequests: Array<fhir.CommunicationRequest>
+): Array<core.Text> {
+  return fhirCommunicationRequests
+    .flatMap(communicationRequest => communicationRequest.payload)
+    .filter(isTruthy)
+    .filter(isContentReferencePayload)
+    .map(payload => resolveReference(fhirBundle, payload.contentReference))
+    .flatMap(list => list.entry)
```

Comment:
Should probably check that `entry`, `item` and `display` are all truthy as the validator doesn't enforce this